app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {})
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = params['metadata']
+        @analysis_id = params['analysis_id']
+      end
+
+      def execute # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength
```

Comment:
**nitpick (blocking):** I'd like you to split up this method into smaller ones to reduce number of rubocop disable and also have it more readable. I will include comments on specific areas that I think can be refactored.

Reply:
Fixed in b78174939f9a80456d3eb8fbc8342885608f718e
Moved `key_transformation` and checking `not_updated_metadata_fields` to their own private methods too, was able to get rid of `cyclomatic` and `perceived complexity` disables.